File: secrets/env.go
package secrets

import (
	"context"
	"fmt"
	"os"
)

// EnvProvider retrieves secrets from environment variables.
type EnvProvider struct{}

// NewEnvProvider creates a new EnvProvider.
func NewEnvProvider() *EnvProvider {
	return &EnvProvider{}
}

// Get retrieves a secret from the environment. Returns an error if the variable is not set.
func (p *EnvProvider) Get(_ context.Context, key string) (string, error) {
	val, ok := os.LookupEnv(key)
	if !ok {
		return "", fmt.Errorf("secrets: environment variable %q not set", key)
	}
	return val, nil
}

// GetOrDefault retrieves a secret from the environment, returning defaultVal if not set.
func (p *EnvProvider) GetOrDefault(_ context.Context, key, defaultVal string) string {
	if val, ok := os.LookupEnv(key); ok {
		return val
	}
	return defaultVal
}

File: src/flag_commons/secrets/env.py
"""Environment-variable secrets provider."""

from __future__ import annotations

import os

from flag_commons.secrets.base import SecretNotFoundError


class EnvProvider:
    """Reads secrets from environment variables.

    A variable that is set but empty returns ``""``, not the default, matching
    Go's ``os.LookupEnv`` semantics.
    """

    def get(self, key: str) -> str:
        try:
            return os.environ[key]
        except KeyError:
            raise SecretNotFoundError(
                f"secrets: environment variable {key!r} not set"
            ) from None

    def get_or_default(self, key: str, default: str) -> str:
        return os.environ.get(key, default)

    def __repr__(self) -> str:
        return "EnvProvider()"
